src/Entity/Comment.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace App\Entity;
+
+use DateTime;
+use DateTimeInterface;
+use Doctrine\ORM\Mapping as ORM;
+use JMS\Serializer\Annotation\Groups;
+use JMS\Serializer\Annotation as Serializer;
+use Swagger\Annotations as SWG;
+
+/**
+ * @ORM\HasLifecycleCallbacks
+ * @ORM\Entity(repositoryClass="App\Repository\CommentRepository")
+ * @Serializer\ExclusionPolicy("all")
+ */
+class Comment
+{
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     * @Serializer\Expose()
+     * @Groups({"Comment"})
+     * @SWG\Property()
+     */
+    private $id;
+
+    /**
+     * Activity id.
+     * @ORM\ManyToOne(targetEntity="Activity")
+     * @ORM\JoinColumn(onDelete="CASCADE")
+     * @Serializer\Expose()
+     * @SWG\Property()
+     */
+    private $activity;
+
+    /**
+     * User id.
+     * @ORM\ManyToOne(targetEntity="User")
+     * @ORM\JoinColumn(onDelete="CASCADE")
+     * @Serializer\Expose()
+     * @Groups({"Comment"})
+     * @SWG\Property()
+     */
+    private $user;
+
+    /**
+     * Activity comment.
+     * @ORM\Column(type="text")
+     * @Serializer\Expose()
+     * @Groups({"Comment"})
+     * @SWG\Property()
+     */
+    private $comment;
+
+    /**
+     * Comment id linked to.
+     * @ORM\Column(type="integer")
+     * @Serializer\Expose()
+     * @Groups({"Comment"})
+     * @SWG\Property()
+     */
+    private $reply;
```

Comment:
Is this the id of the comment to which you reply or the id of the reply one?

Anyway, this should be a relation OneToMany self referenced https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/association-mapping.html#one-to-many-self-referencing

Reply:
You didn't answer the question. As the name creates uncertainty, please rename it to `$parent`.